required/latex-lab/latex-lab-math.dtx

```diff
@@ -129,38 +164,274 @@
 %   \end{itemize}
 % \end{function}
 % 
+% \section{Math tagging}
+% 
+% The tagging code has to handle 
+% \begin{itemize}
+% \item the embedding into the surrounding. This means 
+%   \begin{itemize} 
+%     \item closing and reopening MC-chunks 
+%     \item closing and reopening text/P-structures
+%     \item handling interferences of the tagging code with penalties and spacing.
+%   \end{itemize}
+% \item the actual tagging which means to do some or all of the following tasks:
+%   \begin{itemize}
+%     \item setup content for an associated source file
+%     \item setup content for an associated mathml file 
+%     \item setup content for the /Alt key
+%     \item setup content for the /ActualText key
+%     \item setup attributes
+%     \item add associated files     
+%     \item add a Formula structure
+%     \item surround subparts (e.g. lines) with Formula sub structures
+%          (perhaps with their own set of additional content)
+%     \item surround elements of the equation with mathml structure elements 
+%          (currently only luatex with luamml)
+%    \end{itemize}
+%  \end{itemize}
+%  
+%  \subsection{Inline math}
+%  
+%  The embedding code is added through
+%  the sockets 
+%   \begin{itemize}
+%    \item |tagsupport/math/inline/begin|                
+%    \item |tagsupport/math/inline/end|
+%   \end{itemize}  
+%  The sockets simply push and pop the MC currently. Without
+%  tagging they use the noop-plug. 
+%  
+%  The actual tagging is in done through the sockets
+%   \begin{itemize}
+%    \item |tagsupport/math/inline/formula/begin|
+%     This socket takes the math as argument and its code
+%     should output it for typesetting. Without tagging the 
+%     socket uses the identity plug.
+%     The |default| plug of the socket calls this three internal sockets
+%     for the tagging support:
+%      \begin{itemize}
+%      \item |tagsupport/math/content| This should set up the various
+%      content variables (empty variables are ignored by the structure code
+%       and so can be used to suppress a setting).
+%      \item |tagsupport/math/struct/begin| This calls \cs{tag_struct_begin:n}.
+%      It should also write the associated files if needed.
+%      \item |tagsupport/math/substruct/begin| this handles subparts.
+%      TODO: does it really make sense in inline math to have that?? 
+%      \end{itemize}
+%    \item |tagsupport/math/inline/formula/end|
+%      This socket ends the formula structure(s). The |default|
+%      plug calls these internal sockets:
+%       \begin{itemize}
+%        \item |tagsupport/math/substruct/end|
+%        \item |tagsupport/math/struct/end|      
+%       \end{itemize}
+%   \end{itemize} 
+%    
+%  \subsection{Display math}
+%    
+%  \textit{to be written}
+%    
+%  \subsection{Associated Files}
+%  
+%  The current code allows to attach two type of associated files to the 
```

Comment:
%  The current code allows the attachment of two types of associated file to the